doc/lua_api.txt

```diff
@@ -2360,12 +2360,69 @@ Elements
 * Note: this **must** be before the element is defined.
 * See [Styling Formspecs].
 
-
 ### `style_type[<type>;<prop1>;<prop2>;...]`
 
 * Sets the style for all elements of type `type` which appear after this element.
 * See [Styling Formspecs].
 
+**Note**: do _not_ use a element name starting with `key_` or the name `mouse_event`;
+these names are reserved to pass key press and mouse events to formspecs!
+
+Special Formspec Events
+-----------------------
+
+Formspecs return two different special events not originating from any element:
+`key_event` and `mouse_event`. Both events give JSON that can be translated into
+a table using `minetest.parse_json`.
```

Comment:
Couldn't the client send invalid JSON (or just send `null`) and crash the server?

Reply:
Does invalid JSON crash the server, or does the function just return `nil` if it's invalid?  But yeah, I should recommend making sure it gives correct output.

I don't feel like adding a new `explode_key_event` and an `explode_mouse_event` since they seem kind of redundant, but if I have to, I suppose I will.

Reply:
Invalid JSON [returns `nil` as well as logging an error](https://github.com/minetest/minetest/blob/9284313d17ba906de449c6d39ef3ddacb6c73791/doc/lua_api.txt#L5135).

Reply:
Well, that settles it.  I'll work on `explode_key_event` and `explode_mouse_event` then.